Client/game_sa/CHudSA.cpp

```diff
@@ -43,6 +43,9 @@ void CHudSA::Disable(bool bDisabled)
     else
         MemPut<BYTE>(FUNC_Draw, 0x80);
 
+    for (const auto& component : m_HudComponentMap)
+        SetComponentVisible(component.first, !IsComponentVisible(component.first));
```

Comment:
It looks like `isPlayerHudComponentVisible` will return true for a hidden component after `showhud`

Reply:
Hmm, do you think a copy of the components map would be good here?
To save status of each component and then restore it on the next call